packages/webview_flutter/webview_flutter_wkwebview/lib/src/v4/src/webkit_webview_controller.dart

```diff
@@ -17,14 +17,59 @@ import '../../foundation/foundation.dart';
 import '../../web_kit/web_kit.dart';
 import 'webkit_proxy.dart';
 
+/// Media types that can require a user gesture to begin playing.
+///
+/// See [WebKitWebViewControllerCreationParams.mediaTypesRequiringUserAction].
+enum PlaybackMediaTypes {
+  /// No media types require a user gesture to begin playing.
+  none,
+
+  /// A media type that contains audio.
+  audio,
+
+  /// A media type that contains video.
+  video,
+
+  /// All media types require a user gesture to begin playing.
+  all;
+
+  WKAudiovisualMediaType _toWKAudiovisualMediaType() {
+    switch (this) {
+      case PlaybackMediaTypes.none:
+        return WKAudiovisualMediaType.none;
+      case PlaybackMediaTypes.audio:
+        return WKAudiovisualMediaType.audio;
+      case PlaybackMediaTypes.video:
+        return WKAudiovisualMediaType.video;
+      case PlaybackMediaTypes.all:
+        return WKAudiovisualMediaType.all;
+    }
+  }
+}
+
 /// Object specifying creation parameters for a [WebKitWebViewController].
 @immutable
 class WebKitWebViewControllerCreationParams
     extends PlatformWebViewControllerCreationParams {
   /// Constructs a [WebKitWebViewControllerCreationParams].
   WebKitWebViewControllerCreationParams({
     @visibleForTesting this.webKitProxy = const WebKitProxy(),
-  }) : _configuration = webKitProxy.createWebViewConfiguration();
+    Set<PlaybackMediaTypes>? mediaTypesRequiringUserAction,
```

Comment:
Should we make this parameter non-nullable and defaults to empty set?

Reply:
The current api makes all playback types require user action by default, so I don't want to change that. But I did update this to have a default value instead of being nullable.

https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter/lib/src/webview.dart#L95